packages/logs/src/domain/trackConsoleLogs.ts

```diff
@@ -0,0 +1,49 @@
+import type { ClocksState } from '@datadog/browser-core'
+import { instrumentMethodAndCallOriginal, clocksNow, formatConsoleParameters } from '@datadog/browser-core'
+import type { StatusType } from './logger'
+
+export const ApiName = {
+  log: 'log',
+  debug: 'debug',
+  info: 'info',
+  warn: 'warn',
+} as const
+
+export const LogStatus = {
+  log: 'info',
+  debug: 'debug',
+  info: 'info',
+  warn: 'warn',
+} as const
+
+export type ApiNameType = typeof ApiName[keyof typeof ApiName]
+
+export interface ConsoleLog {
+  message: string
+  status: StatusType
+  startClocks: ClocksState
+}
+
+export function trackConsoleLogs(apis: ApiNameType[], callback: (consoleLog: ConsoleLog) => void) {
+  const proxies = apis.map((api) => startConsoleProxy(api, callback))
+  return {
+    stop: () => {
+      proxies.forEach((proxy) => proxy.stop())
+    },
+  }
+}
+
+/* eslint-disable no-console */
+function startConsoleProxy(api: ApiNameType, callback: (consoleLog: ConsoleLog) => void) {
+  return instrumentMethodAndCallOriginal(console, api, {
+    after: (...params: unknown[]) => callback(buildLogFromParams(params, api)),
+  })
+}
+
+function buildLogFromParams(params: unknown[], api: ApiNameType): ConsoleLog {
+  return {
+    message: [`console ${api}:`, ...params].map((param) => formatConsoleParameters(param)).join(' '),
+    status: LogStatus[api],
+    startClocks: clocksNow(),
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why handling the date here?

Reply:
To have the log build in a single place. `buildConsoleLog` would be a better name. Wdyt?

Reply:
> To have the log build in a single place

not sure to follow you, FMU other attributes are added [here](https://github.com/DataDog/browser-sdk/blob/main/packages/logs/src/boot/logsPublicApi.ts#L103) and [here](https://github.com/DataDog/browser-sdk/blob/main/packages/logs/src/boot/startLogs.ts#L120).

Since among those attributes, the date is already added, I don't see the reasoning for specifically handling it while retrieving attributes from the console api call. 

what am I missing?

Reply:
Okay, I was confused by the date of [rawError](https://github.com/DataDog/browser-sdk/blob/72043747e96c2698290f4f31acc2a8982b81402a/packages/logs/src/boot/startLogs.ts#L70). 
I'll add a comment to clarify it.